src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/Results.tsx

```diff
@@ -94,34 +95,36 @@ export const Results: React.FC<Props> = ({
 
   return (
     <View style={styles.container} testID="tripSearchContentView">
-      {availableTripPatterns.map((tripPattern, i) => (
-        <Fragment key={tripPattern.key}>
-          <DayLabel
-            departureTime={tripPattern.expectedStartTime}
-            previousDepartureTime={tripPatterns[i - 1]?.expectedStartTime}
-          />
-          {newTravelSearchEnabled ? (
-            <ResultItem
-              tripPattern={tripPattern}
-              onDetailsPressed={() => {
-                onDetailsPressed(tripPattern, i);
-              }}
-              searchTime={searchTime}
-              testID={'tripSearchSearchResult' + i}
-              resultNumber={i + 1}
-            />
-          ) : (
-            <ResultItemOld
-              tripPattern={tripPattern}
-              onDetailsPressed={() => {
-                onDetailsPressed(tripPattern, i);
-              }}
-              searchTime={searchTime}
-              testID={'tripSearchSearchResult' + i}
+      {tripPatterns
+        .filter((tp) => !getIsTooLateToBookTripPattern(tp, now))
```

Comment:
Seems like this function is only used here, consider move it to an internal function, instead of in `utils/.`, and consider renaming it as `isLateToBook`

Reply:
We agreed to have it on utils for future reusability.